ouroboros-consensus/src/Ouroboros/Consensus/Protocol/PBFT.hs

```diff
@@ -105,15 +106,21 @@ forgePBftFields PBftIsLeader{..} toSign = do
 
 data PBftLedgerView c = PBftLedgerView {
     -- | ProtocolParameters: map from genesis to delegate keys.
-    pbftDelegates :: Bimap (PBftVerKeyHash c) (PBftVerKeyHash c)
+    pbftDelegates :: !(Bimap (PBftVerKeyHash c) (PBftVerKeyHash c))
   }
   deriving (Generic)
 
 deriving instance PBftCrypto c => NoUnexpectedThunks (PBftLedgerView c)
   -- use generic instance
 
+deriving instance Eq (PBftVerKeyHash c) => Eq (PBftLedgerView c)
 deriving instance Show (PBftVerKeyHash c) => Show (PBftLedgerView c)
 
+instance (Serialise (PBftVerKeyHash c), Ord (PBftVerKeyHash c))
+      => Serialise (PBftLedgerView c) where
+  encode (PBftLedgerView ds) = encode (Bimap.toList ds)
+  decode = PBftLedgerView . Bimap.fromList <$> decode
```

Comment:
@edsko Should the keys and values of the `Bimap` be forced here?

Note that when deserialising the `PBftLedgerView`s we don't get any sharing between the snapshots.